policy/worker.go

```diff
@@ -103,6 +103,12 @@ func (w *Worker) HandlePolicy(ctx context.Context, p *Policy) {
 
 	// Unblock winning handler and cancel the others.
 	for _, handler := range checks {
+
+		// If the handler has exited do not try and send to the proceedCh as
+		// there is not a receiver and we will block the whole execution.
+		if handler.exited {
```

Comment:
does this need a lock?
or... select default?
or... close channel? 
or ... nothing? too much caffeine, feel free to ignore me.